app/models/user.rb

```diff
@@ -0,0 +1,21 @@
+class User < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  validates :name, presence: true
+  validates :photo, presence: true
+  validates :bio, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  before_validation :set_timestamps
+
+  def set_timestamps
+    self.created_at ||= Time.current
+    self.updated_at = Time.current
+  end
+
+  def recent_posts
+    posts.where(author_id: id).order(created_at: :desc).limit(3)
+  end
```

Comment:
- [ Required ] I think it would be nice if you ensure that all custom methods in your models are also tested. Please add a unit test for the `recent_posts` method in the `User` model, since this is a project requirement.

## Example code suggestion
```ruby
  it 'should return a users last 3 posts' do
    user = User.create(
      name: 'Felix',
      bio: 'Software Developer',
      photo: 'http://hello.com/org.png',
      posts_counter: 0
    )
    10.times.collect do
      Post.create(
        title: 'Random title',
        text: 'Hello, world',
        author_id: user.id,
        likes_counter: 0,
        comments_counter: 0
      )
    end
    expect(user.recent_posts.length).to eq(3)
  end
```

## Project requirements
![Screenshot from 2023-11-18 13-10-59](https://github.com/paulshaun-kdtk/blog-posts/assets/16392046/8ae2c1e5-448c-4894-8785-bb8a0b3ece2b)